This is part of a changeset required for the port to build with CMake 3.0.
See patch-cmake__CEGUIMacros.cmake as well.

# HG changeset patch
# User Danduk82 <danduk82@hotmail.com>
# Date 1390065964 -3600
# Branch v0-8
# Node ID 320ecd58eb1b3d673eee62a913041f643c50d483
# Parent  83954df05b3b1b94ada2137ef5d028de54d11818
A) fixed new CMAKE policy CMP0045 issue in CMakeLists.txt. This feature was introduced recently in this commit: https://gitorious.org/cmake/cmake/commit/73e93400e2efab2096618ff58a5ad68236cd04aa ... Bb) fixed CMAKE problem with cmake/CEGUIMacros.cmake when including the project with ExternalProject_Add() CMAKE feature in a project. I had to escape the '[' and ']' characters because in this way CMAKE was not recognizing those characters. CMAKE version: 2.8.12.
--- ./CMakeLists.txt.orig	2013-12-07 11:37:07.000000000 +0200
+++ ./CMakeLists.txt	2014-08-29 00:28:26.000000000 +0300
@@ -6,6 +6,10 @@
     cmake_policy(SET CMP0017 OLD)
 endif()
 
+if (POLICY CMP0045)
+    cmake_policy(SET CMP0045 OLD)
+endif()
+
 set (CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
 include(CEGUIMacros)
 include(CMakeDependentOption)
